### PR TITLE
Account assignment

### PR DESCRIPTION
**Pull request by rylev**
This adds group to account assignments so that the existing groups (infra and infra-admins) have access to the 3 existing accounts (admin, docs-rs-staging, and dev-desktops-prod).

### CODE REVIEW

**Review by jdno — COMMENTED** [review]
I am wondering if we are not making things to difficult for us. Can we reduce the complexity of our setup by putting a few conventions in place? For example:

- `infra-admins` is by default added to every account.
- For every account, we create a `${account_name}-admin` and `${account_name}-read" group with appropriate permissions sets.

Then we just have to manage users and can generate the assignments automatically from a list of account names. Because I expect that each account will only have 2-3 groups anyways: `infra-admins`, a group for maintainers, and maybe a read-only group.

This does assume that the "admin" access for all accounts looks the same, i.e. that we don't limit `docs-rs-staging` to just ECS and RDS, and `dev-desktop-prod` to just `EC2`. Which raises the bigger question on how we want to protect the accounts against mistakes/malice in the first place. `AdministratorAccess` might be a bit broad, and we probably want/need to create budget limits. The specifics of this are something we should discuss outside of this PR, however, we need to decide whether we want to apply generic admin privileges to all accounts.

**Comment by jdno** [comment]
If we want to keep the flexibility of the current approach, I'm fine merging this to unblock other work and iterate in new pull requests.

**Comment by jdno** [comment]
Also curious what @pietroalbini thinks...

**Comment by pietroalbini** [comment]
I think we still want some of this flexibility, even though in the end I'd expect the configuration to just be:

```terraform
accounts = [
  {
    name = "docs-rs-staging"
    admin = ["infra-admins", "infra", "docs-rs"]
    # view = [], automatically adding admin to this
  },
  {
    name = "docs-rs-prod",
    admin = ["infra-admins"]
    view = ["infra", "docs-rs"] # automatically adding admin to this
  },
]
```

We can iterate on that later though (I might prepare a PR this weekend if it'd be helpful).

**Comment by rylev** [comment]
I think I'd like to keep the flexibility. I like @pietroalbini's idea of abstracting over things with "admin" and "view" groupings. I can work on that (still trying to get more and more comfortable with HCL). But I think we should merge this first.

**Review by jdno — APPROVED** [review]
Ok, in that case I'm fine with merging this and iterating in other PRs on the remaining questions. :+1:

Thanks, @rylev! <3

**Pull request merged**
